libs/crypto/hash.hpp

```diff
@@ -48,39 +48,39 @@ namespace iroha {
       return SHA3_512(output, input, in_size);
     }
 
-    nonstd::optional<std::vector<uint8_t>> sha3_224(uint8_t *input,
+    inline nonstd::optional<std::vector<uint8_t>> sha3_224(uint8_t *input,
```

Comment:
`SHA3_*` methods call `KeccakWidth1600_Sponge` methods with predefined parameters. It returns 1 if there is an error with predefined parameters, according to [this](https://github.com/gvanas/KeccakCodePackage/blob/master/Constructions/KeccakSponge.inc) file. Maybe we should guarantee that only correct `unsigned char *output, const unsigned char *input, size_t inputByteLen` are passed, and always return `vector<uint8_t>` or `array<uint8_t, >` of required length?